### PR TITLE
Add the ability to omit fields in schema.

### LINKED ISSUE #185: How to ignore a field when write parquet to file
Just want to ask is there any way to ignore a field when write to parquet file like what encoding/json package did when marshal/unmarshal:
```
// Field is ignored by this package.
Field int `json:"-"`

// Field appears in JSON as key "-".
Field int `json:"-,"`
```

ref: https://pkg.go.dev/encoding/json#Marshal

P/s: Thank, this project is great.

### PR DESCRIPTION
**Pull request by Pryz**
Fixes #185 

Also fixing some other failing tests.

### CODE REVIEW

File: schema.go [hunk 1]
```diff
@@ -297,6 +297,7 @@ func structNodeOf(t reflect.Type) *structNode {
 
 func structFieldsOf(t reflect.Type) []reflect.StructField {
```
Comment (Pryz) [inline schema.go:298]: I'm not familiar with structure fields/tags yet so not quite certain this is the place for this kind of change. Tests show that it is working as intended but would like to get some feedback on this.

Comment (achille-roussel) [reply schema.go:298]: I think `appendStructFields` is the bottom-most function collecting struct fields.

What if we just don't append the fields named `-` in `appendStructFields`, would that work?

File: writer_test.go [hunk 2]
```diff
@@ -277,10 +284,10 @@ value 10: R:0 D:0 V:10.0
 		dump: `row group 0
 --------------------------------------------------------------------------------
 owner:              BINARY ZSTD DO:0 FPO:4 SZ:66/57/0.86 VC:2 ENC:DELTA_LENGTH_BYTE_ARRAY ST:[no stats for this column]
-ownerPhoneNumbers:  BINARY GZIP DO:0 FPO:70 SZ:162/112/0.69 VC:3 ENC:DELTA_LENGTH_BYTE_ARRAY,RLE ST:[no stats for this column]
+ownerPhoneNumbers:  BINARY GZIP DO:0 FPO:70 SZ:162/112/0.69 VC:3 ENC:RLE,DELTA_LENGTH_BYTE_ARRAY ST:[no stats for this column]
```
[the remark this answers is not in the document]
Comment (Pryz) [reply writer_test.go:287]: interesting. I don't see where we are installing parquet-tools for the GH runners. Which version are we supposed to be using ?

File: schema.go (hunk 1 above)
Comment (Pryz) [reply schema.go]: better :) thanks

File: writer_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (achille-roussel) [reply writer_test.go:287]: parquet-tools 1.12.2 with openjdk 18

File: schema.go [hunk 3]
```diff
@@ -306,6 +310,9 @@ func structFieldsOf(t reflect.Type) []reflect.StructField {
 			if name != "" {
 				f.Name = name
 			}
+			if name == "-," {
+				f.Name = "-"
+			}
```
[the remark this answers is not in the document]
Comment (Pryz) [reply schema.go]: nop

File: schema.go [hunk 4]
```diff
@@ -673,6 +686,10 @@ func split(s string) (head, tail string) {
 		head = s
 	} else {
 		head, tail = s[:i], s[i+1:]
+
+		if head == "-" && s[len(s)-1] == ',' {
+			head = "-,"
+		}
```
[the remark this answers is not in the document]
Comment (achille-roussel) [reply schema.go]: I'm fine if we don't allow column names to be "-"

[the remark this answers is not in the document]
Comment (achille-roussel) [reply schema.go]: If you think it's worth supporting, maybe move that to the appendSttuctField function? It's a bit unexpected to have it be a special case in split

[the remark this answers is not in the document]
Comment (Pryz) [reply schema.go]: yep, this is what the last commit changed :)

**Review by achille-roussel — APPROVED** [review]
Looking great!

**Pull request merged**